app/Http/Controllers/PostsController.php

```diff
@@ -108,10 +108,13 @@ public function store(PostRequest $request)
     {
         $northstarId = getNorthstarId($request);
 
-        $signup = $this->signups->get($northstarId, $request['campaign_id']);
+        // Get the campaign id from the request by campaign_id or action_id.
+        $campaignId = $request['campaign_id'] ? $request['campaign_id'] : get_campaign_by_action_id($request['action_id'])->id;
```

Comment:
Instead of a `get_campaign_by_action_id` helper, can we make this a static method on the `Action` model?

Reply:
Or actually, this should probably be on the `Campaign` model since that's what we're fetching, e.g.:

```php
$campaignId = $request['campaign_id'] ?: Campaign::fromActionId($request['action_id'])->id;
```

Reply:
oh yes, I like that a lot better!